tests/components/mysensors/test_init.py

```diff
@@ -22,154 +24,166 @@
+            },
+            1,
+            True,
+            {
+                CONF_GATEWAY_TYPE: CONF_GATEWAY_TYPE_SERIAL,
+                CONF_DEVICE: "COM5",
+                CONF_PERSISTENCE_FILE: "bla.json",
+                CONF_BAUD_RATE: 57600,
+                CONF_VERSION: "2.3",
+            },
+        ),
+        (
+            {
+                DOMAIN: {
+                    CONF_GATEWAYS: [
+                        {
+                            CONF_DEVICE: "127.0.0.1",
+                            CONF_PERSISTENCE_FILE: "blub.pickle",
+                            CONF_BAUD_RATE: 115200,
+                            CONF_TCP_PORT: 343,
+                        }
+                    ],
+                    CONF_VERSION: "2.4",
+                    CONF_PERSISTENCE: False,
+                    CONF_RETAIN: False,
+                }
+            },
+            1,
+            True,
+            {
+                CONF_GATEWAY_TYPE: CONF_GATEWAY_TYPE_TCP,
+                CONF_DEVICE: "127.0.0.1",
+                CONF_PERSISTENCE_FILE: "blub.pickle",
+                CONF_TCP_PORT: 343,
+                CONF_VERSION: "2.4",
+            },
+        ),
+        (
+            {
+                DOMAIN: {
+                    CONF_GATEWAYS: [
+                        {
+                            CONF_DEVICE: "mqtt",
+                            CONF_BAUD_RATE: 115200,
+                            CONF_TCP_PORT: 5003,
+                            CONF_TOPIC_IN_PREFIX: "intopic",
+                            CONF_TOPIC_OUT_PREFIX: "outtopic",
+                        }
+                    ],
+                    CONF_VERSION: "2.4",
+                    CONF_PERSISTENCE: False,
+                    CONF_RETAIN: False,
+                }
+            },
+            1,
+            True,
+            {
+                CONF_GATEWAY_TYPE: CONF_GATEWAY_TYPE_MQTT,
+                CONF_DEVICE: "mqtt",
+                CONF_VERSION: "2.4",
+                CONF_TOPIC_OUT_PREFIX: "outtopic",
+                CONF_TOPIC_IN_PREFIX: "intopic",
+            },
+        ),
+        (
+            {
+                DOMAIN: {
+                    CONF_GATEWAYS: [
+                        {
+                            CONF_DEVICE: "mqtt",
+                            CONF_PERSISTENCE_FILE: "bla.json",
+                            CONF_BAUD_RATE: 115200,
+                            CONF_TCP_PORT: 5003,
+                        },
+                        {
+                            CONF_DEVICE: "COM6",
+                            CONF_PERSISTENCE_FILE: "bla2.json",
+                            CONF_BAUD_RATE: 115200,
+                            CONF_TCP_PORT: 5003,
+                        },
+                    ],
+                    CONF_VERSION: "2.4",
+                    CONF_PERSISTENCE: False,
+                    CONF_RETAIN: False,
+                }
+            },
+            2,
+            True,
+            [],
```

Comment:
Should this be an empty dict?
```suggestion
            {},
```

Reply:
Good catch! I didn't notice because i also forgot to change the function name *facepalm*